cinema/views.py

```diff
@@ -48,6 +73,29 @@
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            self.queryset = self.queryset.filter(show_time__date=date)
+
+        if movie:
+            movie_ids = [int(str_id) for str_id in movie.split(",")]
+            self.queryset = self.queryset.filter(movie__id__in=movie_ids)
+
+        if self.action == "list":
+            self.queryset = self.queryset.select_related(
+                "cinema_hall",
+                "movie",
+            ).annotate(
+                tickets_available=F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets")
+            )
+
+        return self.queryset.distinct()
```

Comment:
Similar to `MovieViewSet`, in the `get_queryset` method of `MovieSessionViewSet`, consider using a local variable instead of modifying `self.queryset` directly to avoid potential issues with repeated calls.

Reply:
Done